demos/segmentation_demo/main.cpp

```diff
@@ -89,178 +64,141 @@ int main(int argc, char *argv[]) {
+        cv::Mat inImg, resImg, maskImg(outHeight, outWidth, CV_8UC3);
+        std::vector<cv::Vec3b> colors(arraySize(CITYSCAPES_COLORS));
+        for (std::size_t i = 0; i < colors.size(); ++i)
+            colors[i] = {CITYSCAPES_COLORS[i].blue(), CITYSCAPES_COLORS[i].green(), CITYSCAPES_COLORS[i].red()};
+        std::mt19937 rng;
+        std::uniform_int_distribution<int> distr(0, 255);
+        int delay = 1;
+        cv::Size graphSize{static_cast<int>(cap.get(cv::CAP_PROP_FRAME_WIDTH) / 4), 60};
+        Presenter presenter(FLAGS_u, 10, graphSize);
+
+        while (cap.read(inImg)) {
+            if (CV_8UC3 != inImg.type())
+                throw std::runtime_error("BGR (or RGB) image expected to come from input");
+            inferRequest.SetBlob(inName, wrapMat2Blob(inImg));
+            inferRequest.Infer();
+
+            const float * const predictions = inferRequest.GetBlob(outName)->cbuffer().as<float*>();
+            for (int rowId = 0; rowId < outHeight; ++rowId) {
+                for (int colId = 0; colId < outWidth; ++colId) {
+                    std::size_t classId = 0;
+                    if (outChannels == 0) {  // assume the output is already ArgMax'ed
+                        classId = static_cast<std::size_t>(predictions[rowId * outWidth + colId]);
                     } else {
-                        /** Iterating over each class probability **/
-                        for (size_t ch = 0; ch < C; ++ch) {
-                            auto data = output_data[image_stride * image + W * H * ch + W * h + w];
-                            if (data > outArrayProb[h][w]) {
-                                outArrayClasses[h][w] = ch;
-                                outArrayProb[h][w] = data;
+                        float maxProb = -1.0f;
+                        for (int chId = 0; chId < outChannels; ++chId) {
+                            float prob = predictions[chId * outHeight * outWidth + rowId * outWidth + colId];
+                            if (prob > maxProb) {
+                                classId = chId;
+                                maxProb = prob;
                             }
                         }
                     }
+                    while (classId >= colors.size()) {
+                        cv::Vec3b color(distr(rng), distr(rng), distr(rng));
+                        colors.push_back(color);
+                    }
+                    maskImg.at<cv::Vec3b>(rowId, colId) = colors[classId];
                 }
             }
-            /** Dump resulting image **/
-            std::string fileName = "out_" + std::to_string(image) + ".bmp";
-            std::ofstream outFile(fileName, std::ofstream::binary);
-            if (!outFile.is_open()) {
-                throw std::logic_error("Can't open file : " + fileName);
+            cv::resize(maskImg, resImg, inImg.size());
+            resImg = inImg * blending + resImg * (1 - blending);
```

Comment:
Why not `cv::addWeighted`?

Reply:
What's the difference?
Even docs say `The function can be replaced with a matrix expression: dst = src1*alpha + src2*beta + gamma;`.

Reply:
Fair enough. I thought it could've been more efficient, since it's a single operation.